edgelet/edgelet-http/src/util/proxy.rs

```diff
@@ -12,26 +13,48 @@ pub struct MaybeProxyClient {
 }
 
 impl MaybeProxyClient {
-    pub fn new(proxy_uri: Option<Uri>) -> Result<Self, Error> {
-        MaybeProxyClient::create(false, proxy_uri)
+    pub fn new(
+        proxy_uri: Option<Uri>,
+        trust_bundle: Option<PemCertificate>,
+    ) -> Result<Self, Error> {
+        MaybeProxyClient::create(false, proxy_uri, None, trust_bundle)
     }
 
-    fn create(null: bool, proxy_uri: Option<Uri>) -> Result<Self, Error> {
+    pub fn new_with_identity_cert(
+        proxy_uri: Option<Uri>,
+        identity_certificate: PemCertificate,
+        trust_bundle: Option<PemCertificate>,
+    ) -> Result<Self, Error> {
+        MaybeProxyClient::create(false, proxy_uri, Some(identity_certificate), trust_bundle)
+    }
+
+    fn create(
```

Comment:
`fn new_inner` is a more clear name for how it's used.

Reply:
Good suggestion